### ISSUE TITLE
Is it possible to add my own validator, which would return a Promise?

### ISSUE THREAD

**Issue by longuniquename** [ctavan/express-validator#94]
My use case is to check if an email is unique across DB table. Pretty common issue. I use [Bookshelf](http://bookshelfjs.org/) to fetch my data. And it's asynchronous! How can I wrap this check into custom validator?

**Comment by sgarbesi** [comment]
+1

**Comment by joelhdzfdz** [comment]
I don't think returning a promise would be a good idea as it would break the syntactic sugar of concatenation, however providing

**Comment by joelhdzfdz** [comment]
I don't think returning a promise would be good as it would break the syntatic sugar, however providing a success and failure method to the custom validators arguments would suffice to accomplish any kind of async validation such as checking wether a record exists in a database. This would look like 

function(value,success,failure){
    models.Country.find({where : {code : value}}).then(function(){
        success();
    },function(){
        failure();
    });
};

**Comment by joelhdzfdz** [comment]
I don't think returning a promise would be good as it would break the syntatic sugar, however providing a success and failure method to the custom validators arguments would suffice to accomplish any kind of async validation such as checking wether a record exists in a database. This would look like 
```
function(value,success,failure){
    models.Country.find({where : {code : value}}).then(function(){
        success();
    },function(){
        failure();
    });
};
```

**Comment by firehawk895** [comment]
@joelhdzfdz I didn't quite understand the solution.

**Comment by renegare** [comment]
This would be great given the growing number of promise based libraries.

**Comment by rustybailey** [comment]
This feature should now be available in 2.17.0. Please check the documentation for [async validation](https://github.com/ctavan/express-validator#asynchronous-validation) and open a new issue if that functionality doesn't work properly.

**Comment by renegare** [comment]
have not checked but +1, will do though ;)
